hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,373 @@
+   * is an instance of OmKeyInfo. If the table is null, returns (0, 0, 0).
+   *
+   * @param table The table from which to get the count and data size.
+   * @return Three Long values representing the count, unReplicated size,
+   *         and replicated size.
+   * @throws IOException If an I/O error occurs during the table iteration.
+   */
+  private Triple<Long, Long, Long> getTableSizeAndCount(Table table)
+      throws IOException {
+    long count = 0;
+    long unReplicatedSize = 0;
+    long replicatedSize = 0;
+
+    TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator =
+        table.iterator();
+    while (iterator.hasNext()) {
+      Table.KeyValue<String, ?> kv = iterator.next();
+      Iterators.size(iterator);
+      if (kv != null && kv.getValue() != null) {
+        if (kv.getValue() instanceof OmKeyInfo) {
+          OmKeyInfo omKeyInfo = (OmKeyInfo) kv.getValue();
+          unReplicatedSize += omKeyInfo.getDataSize();
+          replicatedSize += omKeyInfo.getReplicatedSize();
+        }
+        count++;  // Increment count for each row
+      }
+    }
+    return Triple.of(count, unReplicatedSize, replicatedSize);
+  }
+
+  /**
+   * Returns a collection of table names that require data size calculation.
+   */
+  public Collection<String> getTablesToCalculateSize() {
+    List<String> taskTables = new ArrayList<>();
+    taskTables.add(OPEN_KEY_TABLE);
+    taskTables.add(OPEN_FILE_TABLE);
+    return taskTables;
+  }
+
+  @Override
+  public String getTaskName() {
+    return "OmTableInsightTask";
+  }
+
+  public Collection<String> getTaskTables() {
+    return new ArrayList<>(reconOMMetadataManager.listTableNames());
+  }
+
+  /**
+   * Read the update events and update the count and sizes of respective object
+   * (volume, bucket, key etc.) based on the action (put or delete).
+   *
+   * @param events Update events - PUT, DELETE and UPDATE.
+   * @return Pair
+   */
+  @Override
+  public Pair<String, Boolean> process(OMUpdateEventBatch events) {
+    Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
+    HashMap<String, Long> objectCountMap = initializeCountMap();
+    HashMap<String, Long> unreplicatedSizeCountMap =
+        initializeUnReplicatedSizeMap();
+    HashMap<String, Long> replicatedSizeCountMap =
+        initializeReplicatedSizeMap();
+    final Collection<String> taskTables = getTaskTables();
+    final Collection<String> sizeRelatedTables = getTablesToCalculateSize();
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, Object> omdbUpdateEvent = eventIterator.next();
+
+      if (!taskTables.contains(omdbUpdateEvent.getTable())) {
+        continue;
+      }
+
+      String countKey = getTableCountKeyFromTable(omdbUpdateEvent.getTable());
+      String unReplicatedSizeKey =
+          getUnReplicatedSizeKeyFromTable(omdbUpdateEvent.getTable());
+      String replicatedSizeKey =
+          getReplicatedSizeKeyFromTable(omdbUpdateEvent.getTable());
```

Comment:
omdbUpdateEvent.getTable() is used at multiple places, pls assign to variable and use.

Reply:
Done!